assets/src/application/ui/calendars/dateDisplay/CalendarDateSwitcher/index.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import classNames from 'classnames';
+import { parseISO } from 'date-fns';
+import { __ } from '@wordpress/i18n';
+
+import { DisplayStartOrEndDate } from '@sharedServices/filterState';
+import { PLUS_ONE_MONTH, PLUS_TWO_MONTHS } from '@sharedConstants/defaultDates';
+import { BiggieCalendarDate, CalendarDateRange } from '../';
+
+interface Props {
+	className?: string;
+	displayDate: DisplayStartOrEndDate;
+	endDate: string;
+	footerText?: string;
+	headerPrefix?: string;
+	headerText?: string;
```

Comment:
what might work better for these text labels is to have a separate object with its own interface like:

```js
CalendarDateLabels {
	headerPast: string; // sale starts
	headerFuture: string; // sale started
	footerPast: string; // sale ends
	footerFuture: string; // sale ended
}
```

then we can create a utility for switching tense based on a supplied date like

```js
const headerText = switchTenseForDate( date, labels.headerPast, labels.headerFuture );
```

and then pass that along to the appropriate calendar component.

this way the `CalendarDateSwitcher` will not only handle the responsibility of selecting what component to use for the date display, but also what labels to use.

what do you think? (i'm open for suggestions)

as well, we can simplify this PR and then figure out the above in https://github.com/eventespresso/event-espresso-core/projects/6#card-35163011 since that's what that potential issue is supposed to cover.

Reply:
let's handle this in that issue, because currently I am trying to fix/ understand failing tests :disappointed: